dataprofiler/profilers/profile_builder.py

```diff
@@ -1374,22 +1400,45 @@ def _update_profile_from_chunk(self, data, sample_size,
         elif isinstance(data, list):
             data = pd.DataFrame(data)
 
-        if len(data.columns) != len(data.columns.unique()):
-            raise ValueError('`StructuredProfiler` does not currently support '
-                             'data which contains columns with duplicate '
-                             'names.')
+        duplicate_cols_given = len(data.columns) != len(data.columns.unique())
+
+        # Error out if schema given does not match schema present in data
+        if self._initialized:
+            mapping_given = dict()
+            for i in range(len(data.columns)):
+                col = data.columns[i]
+                if isinstance(col, str):
+                    col = col.lower()
+                mapping_given.setdefault(col, []).append(i)
```

Comment:
if `mapping_given` is initialized to a `defaultdict(list)`, can just do `mapping_given[col].append(i)`

Reply:
not sure which is more greedy